### PR TITLE
Added Fixed and ratio height to sections and Fill on height

### PR DESCRIPTION
**Pull request by jay18001**

### CODE REVIEW

File: Tests/Cells/BaseBrickCellTests.swift [hunk 1]
```diff
@@ -80,7 +80,37 @@ class BaseBrickCellTests: XCTestCase {
...
+    func testSectionFixedHeight() {
+        brickView.registerBrickClass(LabelBrick.self)
+        
+        let section = BrickSection(height: .Fixed(size: 100), bricks: [
+            LabelBrick(height: .Fixed(size: 50), text: "HELLO WORLD", configureCellBlock: { (cell) in })
+            ])
+        brickView.setSection(section)
+        brickView.layoutSubviews()
+        
+        let cell = brickView.cellForItemAtIndexPath(NSIndexPath(forItem: 0, inSection: 0)) as? BrickSectionCell
```
Comment (rubencagnie) [inline Tests/Cells/BaseBrickCellTests.swift]: so what should be the size of the LabelBrick?

File: Tests/Cells/BaseBrickCellTests.swift [hunk 2]
```diff
@@ -80,7 +80,37 @@ class BaseBrickCellTests: XCTestCase {
...
-
+    
+    func testSectionFixedHeight() {
+        brickView.registerBrickClass(LabelBrick.self)
+        
+        let section = BrickSection(height: .Fixed(size: 100), bricks: [
+            LabelBrick(height: .Fixed(size: 50), text: "HELLO WORLD", configureCellBlock: { (cell) in })
+            ])
+        brickView.setSection(section)
+        brickView.layoutSubviews()
```
Comment (rubencagnie) [inline Tests/Cells/BaseBrickCellTests.swift]: please use `brickView.setupSectionAndLayout(section)`

File: Source/Layout/BrickLayoutSection.swift [hunk 3]
```diff
@@ -85,6 +85,10 @@ internal class BrickLayoutSection {
     /// Width of the section. Can be set by `setSectionWidth`
     /// This is not a calculated property of the frame because in case of horizontal scrolling, the values will be different
     internal private(set) var sectionWidth: CGFloat
+    
+    /// Height of the section.
+    /// This is not a calculated property of the frame because in case of vertical scrolling, the values will be different
+    internal private(set) var sectionHeight: CGFloat?
```
Comment (rubencagnie) [inline Source/Layout/BrickLayoutSection.swift]: Isn't the first attribute in the `attributes` variable, the place to get that from?

File: Source/Layout/BrickLayoutSection.swift [hunk 4]
```diff
@@ -45,7 +45,7 @@ protocol BrickLayoutSectionDataSource: class {
     func width(for index: Int, totalWidth: CGFloat, startingAt origin: CGFloat, in section: BrickLayoutSection) -> CGFloat
 
     /// Returns the size for attributes at a given index
-    func size(for attributes: BrickLayoutAttributes, containedIn width: CGFloat, in section: BrickLayoutSection) -> CGSize
+    func size(for attributes: BrickLayoutAttributes, containedInWidth width: CGFloat, containedInHeight: CGFloat, in section: BrickLayoutSection) -> CGSize
```
Comment (rubencagnie) [inline Source/Layout/BrickLayoutSection.swift]: `containedInWidth width: CGFloat, containedInHeight height: CGFloat`

File: Tests/Cells/BaseBrickCellTests.swift (hunk 1 above)
Comment (jay18001) [inline Tests/Cells/BaseBrickCellTests.swift]: Fixed

File: Tests/Cells/BaseBrickCellTests.swift (hunk 2 above)
Comment (jay18001) [inline Tests/Cells/BaseBrickCellTests.swift]: Fixed

File: Source/Layout/BrickLayoutSection.swift (hunk 3 above)
Comment (jay18001) [inline Source/Layout/BrickLayoutSection.swift]: No, its an estimated hight, I'll updated the comment

File: Source/Layout/BrickLayoutSection.swift (hunk 4 above)
Comment (jay18001) [inline Source/Layout/BrickLayoutSection.swift]: Fixed changed to size

File: Tests/Cells/BaseBrickCellTests.swift [hunk 5]
```diff
@@ -80,7 +80,37 @@ class BaseBrickCellTests: XCTestCase {
         XCTAssertEqual(cell?.topSeparatorLine.backgroundColor, .blueColor())
         XCTAssertNil(cell?.bottomSeparatorLine.superview)
     }
-
+    
+    func testSectionFixedHeight() {
```
Comment (rubencagnie) [inline Tests/Cells/BaseBrickCellTests.swift]: What would happen if the bricks inside a section would exceed the fixed height of a section with a fixed height? I'd like to see a test for that

File: Tests/Cells/BaseBrickCellTests.swift [hunk 6]
```diff
@@ -80,7 +80,37 @@ class BaseBrickCellTests: XCTestCase {
...
+        brickView.registerBrickClass(LabelBrick.self)
+        
+        let section = BrickSection(height: .Fixed(size: 100), bricks: [
```
Comment (rubencagnie) [inline Tests/Cells/BaseBrickCellTests.swift]: Could we also add, that when the first section's height is "fill", the section is the same height as the collection view's height. That would be a great feature for the iPad PDP

**Comment by jay18001** [comment]
This causes a lot of issues. Closing for now

**Pull request closed without merging**